### PR TITLE
Moved `Developers` to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by surajssd**
Closes [issue#542](https://github.com/projectatomic/atomicapp/issues/542)

Moved the `Developers` section from README.md to CONTRIBUTING.md
This makes easy for a developer to find all the information in one place.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,6 +15,58 @@ propose changes to this document in a pull request.
...
+Simply run
+
+```shell
+$ virtualenv venv
+$ . venv/bin/activate
+$ cd atomicapp/
+$ python setup.py develop
+$ pip install -r test-requirements.txt
+$ alias atomicapp=<path-to-venv>/venv/bin/atomicapp
+$ alias sudo='sudo '
```
Comment (cdrage) [inline CONTRIBUTING.md]: imo, this is too complicated and not friendly for development, we added `make install`, can't we just use that?

i'd be open to modify Makefile to add a "virtual" environment variable, etx. `make dev`. 

but I'm against this way. each time a person wants to develop they'd have to setup all these aliases again each time.

Comment (surajssd) [inline CONTRIBUTING.md]: Yes will look into `make install` and try to make changes accordingly.

File: Makefile [hunk 2]
```diff
@@ -15,3 +15,11 @@ syntax-check:
...
 clean:
 	python setup.py clean --all
+
+develop:
+	source `which virtualenvwrapper.sh`; \
+	mkvirtualenv atomicapp; \
+	python setup.py develop; \
+	echo "alias atomicapp=~/.virtualenvs/atomicapp/bin/atomicapp" >> ~/.virtualenvs/atomicapp/bin/postactivate; \
+	echo "alias sudo='sudo '" >> ~/.virtualenvs/atomicapp/bin/postactivate; \
+	echo "unalias atomicapp && unalias sudo" >> ~/.virtualenvs/atomicapp/bin/postdeactivate
```
Comment (rtnpro) [inline Makefile]: Cool :+1:

**Comment by rtnpro** [comment]
@surajssd improve the commit message, and we are good to merge.

The first line of your commit message should be a summary of the fix (preferably, less than 60 chars). Followed by an optional summary, and reference to Github issue.

**Comment by cdrage** [comment]
Let's wait for discussion from @dustymabe @goern or @vpavlin 

I don't think this is the best way to introduce someone to development.

At the moment the current workflow is:
```
git clone projectatomic/atomicapp . && atomicapp
# code away
sudo make install
```

Instead of running this in a virtualenv.

Please don't merge for now.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
 * Perform a [cursory search](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Aprojectatomic%2Fatomicapp)
   to see if a similar issue has already been submitted
 
+## Developers
+
+First of all, fork the [repository](https://github.com/projectatomic/atomicapp) and clone your github repository:
```
Comment (cdrage) [inline CONTRIBUTING.md]: `Fork [atomicapp](https://github.com/projectatomic/atomicapp) and clone your GitHub repo:`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
 * Perform a [cursory search](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Aprojectatomic%2Fatomicapp)
   to see if a similar issue has already been submitted
 
+## Developers
```
Comment (cdrage) [inline CONTRIBUTING.md]: `Development`

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
...
+
+First of all, fork the [repository](https://github.com/projectatomic/atomicapp) and clone your github repository:
+
+```shell
+$ git clone https://github.com/<your-username>/atomicapp
+```
+
+### Install this project for development
```
Comment (cdrage) [inline CONTRIBUTING.md]: `Install`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
...
+
+Goto cloned repo and Simply run following commands
```
Comment (cdrage) [inline CONTRIBUTING.md]: `Go to your cloned repo and run the following commands:`

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
...
+```shell
+$ sudo pip install virtualenvwrapper
+$ cd atomicapp
+$ make develop
+```
+
+Goto the Nuleculized app directory and run
+
+```shell
+$ workon atomicapp
```
Comment (cdrage) [inline CONTRIBUTING.md]: what is `workon?`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
...
+
+Goto cloned repo and Simply run following commands
+
+```shell
+$ sudo pip install virtualenvwrapper
+$ cd atomicapp
+$ make develop
+```
+
+Goto the Nuleculized app directory and run
```
Comment (cdrage) [inline CONTRIBUTING.md]: `Go to your Nulecule app directory and run:`

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
...
+$ workon atomicapp
+$ sudo atomicapp run .
+```
+
+### Build
+```
+docker build -t [TAG] .
+```
+
+Just a call to Docker to package up the application and tag the resulting image.
```
Comment (cdrage) [inline CONTRIBUTING.md]: `Build and tag the resulting image.`

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -15,6 +15,54 @@ propose changes to this document in a pull request.
...
+existing path, it looks for `Nulecule` file there instead of pulling anything.
+
+* `--recursive yes|no` Pull whole dependency tree
+* `--update` Overwrite any existing files
+* `--destination DST_PATH` Unpack the application into given directory instead of current directory
+* `APP` Name of the image containing the application (f.e. `vpavlin/wp-app`)
+* `PATH` Path to a directory with installed (i.e. result of `atomicapp install ...`) app
+
+Action `run` performs `install` prior its own tasks are executed if `APP` is given. When `run` is selected, providers' code is invoked and containers are deployed.
+
```
Comment (cdrage) [inline CONTRIBUTING.md]: This entire part is old, please have a look at #535

File: CONTRIBUTING.md (hunk 7 above)
Comment (surajssd) [inline CONTRIBUTING.md]: `workon` is a `virtualenvwrapper`'s construct. See `virtualenvwrapper` [docs](https://virtualenvwrapper.readthedocs.org/en/latest/)

File: CONTRIBUTING.md (hunk 3 above)
Comment (surajssd) [inline CONTRIBUTING.md]: Acknowledged

File: CONTRIBUTING.md (hunk 4 above)
Comment (surajssd) [inline CONTRIBUTING.md]: Acknowledged :+1:

File: CONTRIBUTING.md (hunk 5 above)
Comment (surajssd) [inline CONTRIBUTING.md]: I don't understand this part.

File: CONTRIBUTING.md (hunk 6 above)
Comment (surajssd) [inline CONTRIBUTING.md]: Acknowledged :+1:

File: CONTRIBUTING.md (hunk 8 above)
Comment (surajssd) [inline CONTRIBUTING.md]: Will take care of it.

File: CONTRIBUTING.md (hunk 10 above)
Comment (surajssd) [inline CONTRIBUTING.md]: Once #535 gets merged maybe those changes would reflect here.

**Comment by surajssd** [comment]
Okay I will wait for others to comment and then we can move ahead.

File: CONTRIBUTING.md (hunk 5 above)
Comment (cdrage) [inline CONTRIBUTING.md]: title `Install this project for development` should be just `Install`

**Comment by dustymabe** [comment]
A few things here. It is much easier for reviews and for reading through change history if you don't move code (or docs in this case) and also change the content in the same commit. It's usually better to copy the code in one commit and then make changes to it in the next.

As for the virtualenv stuff. I don't mind having a section that explains how to use it with `atomicapp` but I'd rather not make it a requirement. I'm not sure having it be part of the Makefile is desirable. The virtualenv stuff should also be in a separate commit. 

To summarize:

- Let's break this up into separate commits
- Let's use descriptive commit messages 
- After you update the PR we'll review again and see where the virtualenv stuff fits in.

Also charlie should be getting some changes in for his PR on the readme and they should go in soon so you may want to wait to update this PR til after that.

**Comment by surajssd** [comment]
First off I will wait for the @cdrage 's PR on README once that gets pulled further I will make respective changes.

About dividing the changes into 'move' and 'change', should this be accompained with new issue as well?

And one suggestion in make file we can make something like 'install for development' with virtualenv and without virtualenv, which ever folks choose viz `make develop` and `make  develop virtualenv` or something with better names!

**Comment by dustymabe** [comment]
> First off I will wait for the @cdrage 's PR on README once that gets pulled further I will make respective changes.

Thanks!

> 
> About dividing the changes into 'move' and 'change', should this be accompained with new issue as well?
> 

No new issue needed. 

> And one suggestion in make file we can make something like 'install for development' with virtualenv and without virtualenv, which ever folks choose viz make develop and make develop virtualenv or something with better names!

perhaps we could go for something like that. I'll review your proposal when you update this PR and we'll see where to go with it.

**Comment by cdrage** [comment]
Hey @surajssd I've fixed this issue with #576 :) Let me know if you'd like anything added to the CONTRIBUTING.md section!

**Comment by surajssd** [comment]
@cdrage :+1: great work on the docs side.
